Adafruit_AMG88xx.cpp

```diff
@@ -254,25 +252,22 @@ void Adafruit_AMG88xx::read(uint8_t reg, uint8_t *buf, uint8_t num) {
 }
 
 void Adafruit_AMG88xx::write(uint8_t reg, uint8_t *buf, uint8_t num) {
+  uint8_t buffer[num + 1];
+  buffer[0] = reg;
 #ifdef I2C_DEBUG
   Serial.print("[$");
   Serial.print(reg, HEX);
   Serial.print("] <- ");
 #endif
-  Wire.beginTransmission((uint8_t)_i2caddr);
-  Wire.write((uint8_t)reg);
   for (int i = 0; i < num; i++) {
-    Wire.write(buf[i]);
+    buffer[i + 1] = buf[i];
 #ifdef I2C_DEBUG
     Serial.print("0x");
     Serial.print(buf[i], HEX);
     Serial.print(", ");
 #endif
   }
-  Wire.endTransmission();
-#ifdef I2C_DEBUG
-  Serial.println();
-#endif
+  i2c_dev->write(buffer, num + 1);
```

Comment:
can also just use the 'prefix' buffer (optional arg) instead of most of this stuff :)

Reply:
neat!